src/hooks/useChart.ts

```diff
@@ -1,46 +1,59 @@
 import { useEffect, useRef, useState } from 'react'
 import { Token } from 'uniswap-xdai-sdk'
 
-import * as am4core from '@amcharts/amcharts4/core'
-import am4themesSpiritedaway from '@amcharts/amcharts4/themes/spiritedaway'
+import * as am4charts from '@amcharts/amcharts4/charts'
 
+import { XYChartProps, drawInformation } from '../components/auction/Charts/XYChart'
 import { PricePointDetails } from '../components/auction/OrderbookChart'
 
-am4core.useTheme(am4themesSpiritedaway)
-
-const useChart = ({
-  baseToken,
-  createChart,
-  data,
-  quoteToken,
-}: {
-  createChart: any
+interface Props {
+  createChart: (props: XYChartProps) => am4charts.XYChart
   data: Maybe<PricePointDetails[]>
   baseToken: Token
   quoteToken: Token
-}) => {
-  const [chart, setChart] = useState(null)
-  const elemRef = useRef(null)
+}
+
+const useChart = (props: Props) => {
+  const { baseToken, createChart, data, quoteToken } = props
+
   const [loading, setLoading] = useState(false)
 
+  const mountPoint = useRef<Maybe<HTMLDivElement>>(null)
+  const chartRef = useRef<Maybe<am4charts.XYChart>>(null)
```

Comment:
Could you please leave a comment in the code that explains this mountPoint logic. 

I would love to understand your solution, given that it was holding us back so long.

Reply:
Sure